src/components/TaskCard/EditTaskCard.jsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import PropTypes from "prop-types";
+import { sanitizeText } from "utils/helpers/sanitizeText.js";
+import { editTask, completeTask } from "store/actions";
+import "components/TaskCard/index.scss";
+import {
+  ICON_COMPLETE,
+  COMPLETE_ICON_ALT_TEXT,
+  ICON_DELETE,
+  DELETE_ICON_ALT_TEXT,
+  KEY_ENTER,
+} from "utils/constant";
+
+const EditTaskCard = ({ id, task, onEditableTasks }) => {
+  const [inputText, setInputText] = useState(task);
+  const [error, setError] = useState("");
+  const dispatch = useDispatch();
+
+  function handleInputChange(event) {
+    setInputText(event.target.value);
+  }
+
+  function onSave() {
+    const sanitizedTask = sanitizeText(inputText);
+
+    if (sanitizedTask === "") {
+      setError("Please add task description");
+      return;
+    }
+
+    dispatch(
+      editTask({
+        id,
+        task: sanitizedTask,
+      })
+    );
+
+    onEditableTasks(false);
```

Comment:
why are we sending `false` on edit?

Reply:
my mistake . I used this one for toggling task editing option.